src/harness/testcases/WINNF_FT_S_PCR_testcase.py

```diff
@@ -179,22 +172,93 @@ def triggerCpasAndRetrievePpaZone(self, ppa_id, ssl_cert, ssl_key):
+
+    Performs the assert to check installationParam present in registrationRequests or
+    conditional registration data and raises an exception.
+    PpaCreationModel requires the input registrationRequests to have 'installationParam'.
+    But this parameter is removed for devices where conditionals are pre-loaded.
+    Adding the 'installationParam' into registrationRequests by taking the corresponding
+    values from conditionalRegistrationData.
+
+    Args:
+      registration_request: A list of individual CBSD registration
+        requests (each of which is itself a dictionary).
+      conditional_registration_data: A list of individual CBSD registration
+        data that need to be preloaded into SAS (each of which is a dictionary).
+        the fccId and cbsdSerialNumber fields are required, other fields are optional
+        but required for ppa reference model.
+
+    Raises:
+      It will throws an exception if the installationParam object and required fields is not found
+    in conditionalRegistrationData and registrationRequests for category B and A devices
+    respectively.
 
-      # Check if the retrieved FAD that has valid PPA zone record matches the PPA ID
-      # that was generated using the admin API.
-      uut_ppa_zone_data = uut_fad.getZoneRecords([lambda x: x['id'] == ppa_id])
-      self.assertEquals(len(uut_ppa_zone_data), 1,
-                        msg='There is no single PPA Zone record matches with PPA ID '
-                            '{0} received from SAS UUT'.format(ppa_id))
-    except KeyError:
-      self.fail('There is an error while filtering PPA Zone record based '
-                'on PPA ID:{0}'.format(ppa_id))
-    else:
-      return uut_ppa_zone_data[0]
+    """
+
+    for device in registration_request:
+      if 'installationParam' not in device:
+        for conditional_params in conditional_registration_data:
+          # Check if FCC_ID+Serial_Number present in registrationRequest
+          # and conditional_params match and add the 'installationParam'.
+          self.assertIn('installationParam', conditional_params,
+                        msg='installationParm Object is not found in REG-Conditionals')
+          if not (conditional_params['fccId'] == device['fccId'] and \
+                  conditional_params['cbsdSerialNumber'] == device['cbsdSerialNumber']):
+            raise Exception('ConfigError:Wrong REG-Conditional data for device is found. '
+                            'Please load the correct REG-Conditional data for the device')
+          else:
+            # The following REG-conditional parameters are required to present in
+            # installationParam Object and cbsdCategory in RegistrationRequest for PPA
+            # reference model to determine PPA contour boundary.
+            # assert that all the needed parameters for PPA are present.
+            if any([conditional_param_name not in conditional_params['installationParam']
```

Comment:
I'm not sure azimuth and beamwidth are technically required. These params are required for Cat B and Cat B requires CPI/conditional reg, but you can always have a Cat A which chooses conditional reg data, and these params are not required for Cat A devices.

Reply:
Agreed.   We have modified the code to fill in 'installationParam' and 'deviceCategory'   from the conditionals if these are not present in registrationRequest.   Considering that some parameters are optional for Cat-A and Cat-B we think it   is a better to have the PPA reference model check for the fields that is   needed for its calculation and fill in wither default values or raise an   exception. Maybe we can have an issue PR raised to check for the necessary   fields in the PPA model. Please let us know if you agree.